account_invoice_qr_code_sepa_payconiq/models/res_partner_bank.py

```diff
@@ -0,0 +1,182 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import base64
+import io
+import logging
+import urllib
+
+import requests
+from PIL import Image
+
+from odoo import _, api, models
+from odoo.tools.image import image_data_uri
+
+PAYCONIQ_URL = "https://payconiq.com/t/1/"
+PAYCONIQ_QR_URL = "https://portal.payconiq.com/qrcode"
+
+_logger = logging.getLogger(__name__)
+
+
+class ResPartnerBank(models.Model):
+
+    _inherit = "res.partner.bank"
+
+    @api.model
+    def _get_available_qr_methods(self):
+        rslt = super()._get_available_qr_methods()
+        rslt.append(("payconiq_qr", _("Payconiq QR"), 19))
+        return rslt
+
+    def _get_payconiq_qr_amount(self, amount):
+        """
+        Payconiq is awaiting an amount in eurocents. So,
+        if the amount is for instance 141.9, reformat to float with
+        two decimals => 141.90, then remove decimal dot.
+        """
+        return "{amount:.2f}".format(amount=amount).replace(".", "")
+
+    def _get_qr_code_generation_params(
+        self,
+        qr_method,
+        amount,
+        currency,
+        debtor_partner,
+        free_communication,
+        structured_communication,
+    ):
+        """
+        https://developer.payconiq.com/online-payments-dock/#creating-the-payconiq-qr-code77
+
+        """
+        if qr_method == "payconiq_qr":
+            return {
+                "payconiq_qr": True,
+                "D": "Payment",
+                "A": self._get_payconiq_qr_amount(amount),
+                "R": structured_communication
+                if structured_communication
+                else free_communication,
+            }
+        return super()._get_qr_code_generation_params(
+            qr_method,
+            amount,
+            currency,
+            debtor_partner,
+            free_communication,
+            structured_communication,
+        )
+
+    def _get_qr_code_url(
+        self,
+        qr_method,
+        amount,
+        currency,
+        debtor_partner,
+        free_communication,
+        structured_communication,
+    ):
+        params = self._get_qr_code_generation_params(
+            qr_method,
+            amount,
+            currency,
+            debtor_partner,
+            free_communication,
+            structured_communication,
+        )
+        if params and params.get("payconiq", False):
+            params.pops("payconiq_qr")
+            return urllib.parse.urlencode(params)
+        return super()._get_qr_code_url(
+            qr_method,
+            amount,
+            currency,
+            debtor_partner,
+            free_communication,
+            structured_communication,
+        )
+
+    def _eligible_for_qr_code(self, qr_method, debtor_partner, currency):
+        if qr_method == "payconiq_qr":
+            return (
+                currency.name == "EUR"
+                and self.acc_type == "iban"
+                and self.sanitized_acc_number[:2] in ["LU"]
```

Comment:
So it's only available for Lux ?

Reply:
Their mechanism is europeanwide but the implementation is available there for the moment.

Reply:
So the company account must be luxembourgish ?